File: src/my_robot_tutorials/src/add_two_ints_server.cpp
#include <ros/ros.h>
#include <rospy_tutorials/AddTwoInts.h>


// functions
bool handle_add_two_ints(rospy_tutorials::AddTwoInts::Request& req, 
	                       rospy_tutorials::AddTwoInts::Response& res)
{
	int result = req.a + req.b;
	ROS_INFO("%d + %d = %d", (int)req.a, (int)req.b, (int)result);
	res.sum = result;
	return true; // service call was successfull
}


// main
int main(int argc, char** argv)
{
	// init node
	ros::init(argc, argv, "add_two_ints_server");
	ros::NodeHandle nh;

	// create the service
	ros::ServiceServer server = nh.advertiseService("/add_two_ints", handle_add_two_ints);

	// keep running server until this node is shutdown
	ros::spin();

}

File: src/my_robot_tutorials/scripts/add_two_ints_server.py
#!/usr/bin/env python
import rospy
from rospy_tutorials.srv import AddTwoInts


# functions
def handle_add_two_ints(req):
	result = req.a + req.b
	rospy.loginfo("Sum of " + str(req.a) + " and " + str(req.b) + " is " + str(result))
	return result


# main
if __name__=='__main__':

	# init  node
	rospy.init_node("add_two_ints_server" )
	rospy.loginfo("Add two ints server node created")

	# create a service (service_name, message_type, callback_function)
	service = rospy.Service("/add_two_ints", AddTwoInts, handle_add_two_ints)
	rospy.loginfo("Service server has been started")

	# keep the service running until the 'add_two_ints_server' is shutdown
	rospy.spin()
	

'''
NOTES: AddTwoInts message type info can be found at:
       http://docs.ros.org/jade/api/rospy_tutorials/html/srv/AddTwoInts.html
       
       The service definition can be found in the file system of this computer;
       the file is located at:
       /opt/ros/kinetic/share/rospy_tutorials/srv/AddTwoInts.srv
'''
